cpp/dolfinx/mesh/topologycomputation.cpp

```diff
@@ -404,38 +404,48 @@ compute_entities_by_key_matching(
   // Initialize local array of entities
   const std::int8_t num_entities_per_cell
       = mesh::cell_num_entities(cell_type, dim);
-  const std::size_t num_vertices_per_entity
-      = mesh::num_cell_vertices(mesh::cell_entity_type(cell_type, dim));
+
+  // For some cells, the num_vertices varies per facet (3 or 4)
+  int max_vertices_per_entity = 0;
+  for (int i = 0; i < num_entities_per_cell; ++i)
```

Comment:
This is a bit ugly - could it be neater to use `get_sub_entities`?

Reply:
I don't think we can use that, because we need something more general (`get_sub_entities` only works from face->edge).

Reply:
Could do:
```
  std::vector<mesh::CellType> entity_type(num_entities_per_cell);
  for (int i = 0; i < num_entities_per_cell; ++i)
    entity_type[i] = mesh::cell_entity_type(cell_type, dim, i);

  int max_vertices_per_entity = std::transform_reduce(
      entity_type.begin(), entity_type.end(), 0,
      [](int a, int b) { return std::max(a, b); }, &mesh::num_cell_vertices)
```
or
```
int max_vertices_per_entity = 0;
 if (dim == 1)
   max_vertices_per_entity = 2;
 else if (dim == 2 and cell_type == mesh::CellType::prism)
   max_vertices_per_entity = 4;
 else if (dim == 2)
   max_vertices_per_entity = mesh::num_cell_vertices(mesh::cell_entity_type(cell_type, dim, 0);
```

I think I'm inclined to stick with the original version...